crates/pathfinder/src/sequencer.rs

```diff
@@ -137,13 +159,16 @@ impl Client {
                 )?;
             Ok(value)
         } else {
-            let error = from_value::<reply::starknet::Error>(json_val)?;
-            Err(anyhow::Error::new(error))
+            let e = from_value::<StarknetError>(json_val)?;
```

Comment:
```suggestion
            let e = serde_json::from_value::<StarknetError>(json_val)?;
```
I think having the `serde_json` improves readability -- I wasn't sure what `from_value` was doing at first glance. And later on you do use `serde_json::from_str` which I think is better. Probably occurs in other places as well.

Reply:
I have the same feeling - I kept on keeping `serde_json` elsewhere :thinking: . I cannot recall why I did this though.